packages/nextjs/src/config/webpack.ts

```diff
@@ -143,6 +144,33 @@ async function addSentryToEntryProperty(
   return newEntryProperty;
 }
 
+/**
+ * Search the project directory for a valid user config file for the given platform, allowing for it to be either a
+ * TypeScript or JavaScript file.
+ *
+ * @param projectDir The root directory of the project, where the file should be located
+ * @param platform Either "server" or "client", so that we know which file to look for
+ * @returns The name of the relevant file. If no file is found, this method throws an error.
+ */
+function getUserConfigFile(projectDir: string, platform: 'server' | 'client'): string {
+  let configFile;
+
+  const possibilities = [`sentry.${platform}.config.ts`, `sentry.${platform}.config.js`];
+
+  for (const filename of possibilities) {
+    if (fs.existsSync(path.resolve(projectDir, filename))) {
+      configFile = filename;
+      break;
+    }
+  }
+
+  if (!configFile) {
+    throw new Error(`Cannot find ${possibilities[0]} or ${possibilities[1]} in ${projectDir}`);
```

Comment:
I was torn here, about whether to throw an error or merely `console.error` a message. OTOH, this means we can break people's builds. OTOH, that can only happen if they're using `withSentryConfig` but don't have the accompanying config files (IOW, clearly intending to use Sentry, but doing it wrong). I settled on throwing an error because it forces people to fix it right away, which means they can't fall into a "the SDK mysteriously isn't working and I'm conveniently ignoring the error message which would tell me why" state.

Open to other opinions here, though.

Reply:
I guess the only other possibility is that they are using `withSentryConfig` to upload source maps, but are initializing Sentry some other way, but the workaround is just to use our webpack plugin by itself so I can’t really see that happening. 

My opinion is that we should throw an error.

Reply:
> the workaround is just to use our webpack plugin by itself

Or just to throw an empty `sentry.server.config.js` file into their folder, I suppose. But I agree, that's a weird edge case. An error it is, then. Thanks for the second opinion. 🙂